### PR TITLE
Feature/377 fontys gitlab datasource to wizard

### PR DESCRIPTION
**Pull request by TimSnoek123**
## Description

Added the functionality to search for repositories using the Fontys Gitlab, private projects for gitlab should also work. @rubenb994 can you maybe test if gitlab still works? I don't have the client secret etc...

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I did not update API Controllers, if I did, I added/changed Postman tests
-   [X] I have performed a self-review of my own code
-   [X] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [X] New and existing unit tests pass locally with my changes
-   [X] I updated the changelog with an end-user readable description
-   [X] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

1. Get a user token, clientid and clientsecret from the Fontys Gitlab (in your user settings).
2. Open swagger and check all 3 wizard endpoints using the token and other meta data specific to the request.
3. Check if gitlab works with private and public projects.

## Link to issue

Closes: #377

### CODE REVIEW

**Comment by TimSnoek123** [comment]
Currently the integration tests are executing really slowly compared to develop. This also causes them to fail, don't know why this happens. :(

**Comment by niraymak** [comment]
Worked like a charm. I would like to check the GitHub private flow again. Didn't get it working. Will check next week together with @RubenFricke

**Comment by TimSnoek123** [comment]
Don't know if the github private flow works yet tbh we mostly focused on gitlab.

File: API/appsettingsapi.Development.json [hunk 1]
```diff
@@ -32,13 +32,20 @@
...
+                "ClientSecret": "627a2ce856d47c95878e1dce2ada57ec135d3021fe916b1590f4cee6ddae37d7",
+                "RedirectUri": "https://localhost:5001/api/wizard/oauth/callback/6a73f5ee-e7a5-4ef5-b874-0da0993d6f13",
+                "OauthUrl": "https://git.fhict.nl/oauth/authorize?client_id={0}&redirect_uri={1}&response_type=code&scope=read_repository+read_user"
             }
         },
         "RabbitMQ": {
             "Hostname": "localhost",
-            "Username": "test",
-            "Password": "test"
+            "Username": "guest",
```
Comment (niraymak) [inline API/appsettingsapi.Development.json:47]: please change back to "test", "test"

File: API/appsettingsapi.Development.json [hunk 2]
```diff
@@ -32,13 +32,20 @@
...
                 "ClientSecret": "",
-                "RedirectUri": "https://localhost:5001/api/wizard/oauth/callback/66de59d4-5db0-4bf8-a9a5-06abe8d3443a"
+                "RedirectUri": "https://localhost:5001/api/wizard/oauth/callback/66de59d4-5db0-4bf8-a9a5-06abe8d3443a",
+                "OauthUrl": "https://gitlab.com/oauth/authorize?client_id={0}&redirect_uri={1}&response_type=code&scope=read_repository+read_user"
+            },
+            "FontysGitlab": {
+                "ClientId": "536f6b353a9ecea950f69e9141ce87579376c279d7201c42099266a768148632",
+                "ClientSecret": "627a2ce856d47c95878e1dce2ada57ec135d3021fe916b1590f4cee6ddae37d7",
+                "RedirectUri": "https://localhost:5001/api/wizard/oauth/callback/6a73f5ee-e7a5-4ef5-b874-0da0993d6f13",
+                "OauthUrl": "https://git.fhict.nl/oauth/authorize?client_id={0}&redirect_uri={1}&response_type=code&scope=read_repository+read_user"
```
Comment (niraymak) [inline API/appsettingsapi.Development.json:42]: I think this should be removed, right? Or can it be used for development?

@RubenFricke we should add FHICT GitLab API keys to staging / production

**Review by niraymak — CHANGES_REQUESTED** [review]
Thanks for your implementation 💯 

Can you please change RabbitMQ appsettings back to docker credentials. 
Also please remove GitLab API key.

File: API/appsettingsapi.Development.json [hunk 3]
```diff
@@ -32,13 +32,20 @@
             "Gitlab": {
                 "ClientId": "",
                 "ClientSecret": "",
-                "RedirectUri": "https://localhost:5001/api/wizard/oauth/callback/66de59d4-5db0-4bf8-a9a5-06abe8d3443a"
+                "RedirectUri": "https://localhost:5001/api/wizard/oauth/callback/66de59d4-5db0-4bf8-a9a5-06abe8d3443a",
```
Comment (niraymak) [inline API/appsettingsapi.Development.json:35]:
```suggestion
                "RedirectUri": "",
```

File: API/appsettingsapi.Development.json [hunk 4]
```diff
@@ -32,13 +32,20 @@
...
+                "OauthUrl": "https://gitlab.com/oauth/authorize?client_id={0}&redirect_uri={1}&response_type=code&scope=read_repository+read_user"
```
Comment (niraymak) [inline API/appsettingsapi.Development.json:36]:
```suggestion
                "OauthUrl": ""
```

File: API/appsettingsapi.Development.json [hunk 5]
```diff
@@ -32,13 +32,20 @@
...
+            },
+            "FontysGitlab": {
+                "ClientId": "536f6b353a9ecea950f69e9141ce87579376c279d7201c42099266a768148632",
```
Comment (niraymak) [inline API/appsettingsapi.Development.json:39]:
```suggestion
                "ClientId": "",
```

File: API/appsettingsapi.Development.json [hunk 6]
```diff
@@ -32,13 +32,20 @@
...
+                "ClientSecret": "627a2ce856d47c95878e1dce2ada57ec135d3021fe916b1590f4cee6ddae37d7",
```
Comment (niraymak) [inline API/appsettingsapi.Development.json:40]:
```suggestion
                "ClientSecret": "",
```

File: API/appsettingsapi.Development.json [hunk 7]
```diff
@@ -32,13 +32,20 @@
...
+                "RedirectUri": "https://localhost:5001/api/wizard/oauth/callback/6a73f5ee-e7a5-4ef5-b874-0da0993d6f13",
```
Comment (niraymak) [inline API/appsettingsapi.Development.json:41]:
```suggestion
                "RedirectUri": "",
```

File: API/appsettingsapi.Development.json (hunk 1 above)
Comment (niraymak) [reply API/appsettingsapi.Development.json:47]:
```suggestion
            "Username": "test",
```

File: API/appsettingsapi.Development.json [hunk 8]
```diff
@@ -32,13 +32,20 @@
...
+                "OauthUrl": "https://git.fhict.nl/oauth/authorize?client_id={0}&redirect_uri={1}&response_type=code&scope=read_repository+read_user"
             }
         },
         "RabbitMQ": {
             "Hostname": "localhost",
-            "Username": "test",
-            "Password": "test"
+            "Username": "guest",
+            "Password": "guest"
```
Comment (niraymak) [inline API/appsettingsapi.Development.json:48]:
```suggestion
            "Password": "test"
```

**Review by niraymak — CHANGES_REQUESTED** [review]

**Review by niraymak — APPROVED** [review]

**Review by RubenFricke — APPROVED** [review]
Good work guys!

**Pull request merged**